iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/SV.java

```diff
@@ -84,132 +74,341 @@ public double getA() {
+    @Deprecated
+    public SV otherSideY1Y2(Line l) {
+        throw new PowsyblException("Deprecated. Not used anymore");
     }
 
     public SV otherSide(TwoWindingsTransformer twt) {
-        return otherSide(getR(twt), getX(twt), getG(twt), getB(twt), getRho(twt));
+        return otherSide(getR(twt), getX(twt), getG(twt), getB(twt), 0.0, 0.0, getRho(twt), getAlpha(twt));
     }
 
-    public SV otherSide(Line l) {
-        return otherSide(l.getR(), l.getX(), l.getG1() + l.getG2(), l.getB1() + l.getB2(), 1);
+    public SV otherSide(TwoWindingsTransformer twt, boolean splitShuntAdmittance) {
+        if (splitShuntAdmittance) {
+            return otherSide(getR(twt), getX(twt), getG(twt) * 0.5, getB(twt) * 0.5, getG(twt) * 0.5, getB(twt) * 0.5, getRho(twt), getAlpha(twt));
+        } else {
+            return otherSide(getR(twt), getX(twt), getG(twt), getB(twt), 0.0, 0.0, getRho(twt), getAlpha(twt));
+        }
     }
 
-    public SV otherSideY1Y2(Line l) {
-        return otherSide(l.getR(), l.getX(), l.getG1(), l.getB1(), l.getG2(), l.getB2(), 1);
+    // Has been modified
+    // Before: otherSide(dl.getR(), dl.getX(), dl.getG() / 2.0, dl.getB() / 2.0, dl.getG() / 2.0, dl.getB() / 2.0, 1.0, 0.0);
+    public SV otherSide(DanglingLine dl) {
+        return otherSide(dl.getR(), dl.getX(), dl.getG(), dl.getB(), 0.0, 0.0, 1.0, 0.0);
     }
 
-    public SV otherSide(DanglingLine dl) {
-        return otherSide(dl.getR(), dl.getX(), dl.getG() / 2.0, dl.getB() / 2.0, dl.getG() / 2.0, dl.getB() / 2.0, 1);
+    public SV otherSide(DanglingLine dl, boolean splitShuntAdmittance) {
+        if (splitShuntAdmittance) {
+            return otherSide(dl.getR(), dl.getX(), dl.getG() * 0.5, dl.getB() * 0.5, dl.getG() * 0.5, dl.getB() * 0.5, 1.0, 0.0);
+        } else {
+            return otherSide(dl.getR(), dl.getX(), dl.getG(), dl.getB(), 0.0, 0.0, 1.0, 0.0);
+        }
     }
 
+    /**
+     * @deprecated Not used anymore. Use
+     * {@link SV#otherSideP(double, double, double, double, double, double, double, double)}
+     */
+    @Deprecated
     public double otherSideP(double r, double x, double g1, double b1, double g2, double b2, double rho) {
-        Complex z = new Complex(r, x); // z=r+jx
-        Complex y1 = new Complex(g1, b1); // y1=g1+jb1
-        Complex y2 = new Complex(g2, b2); // y2=g2+jb2
-        Complex s2 = computeS2(y1, y2, z, rho);
-        return -s2.getReal();
+        throw new PowsyblException("Deprecated. Not used anymore");
     }
 
+    public double otherSideP(double r, double x, double g1, double b1, double g2, double b2, double rho, double alpha) {
+        LinkData.BranchAdmittanceMatrix adm = LinkData.calculateBranchAdmittance(r, x, 1 / rho, -alpha, 1.0, 0.0,
+            new Complex(g1, b1), new Complex(g2, b2));
+        return otherSideP(adm);
+    }
+
+    // Has been modified
+    // Before: otherSideP(dl.getR(), dl.getX(), dl.getG() / 2.0, dl.getB() / 2.0, dl.getG() / 2.0, dl.getB() / 2.0, 1.0, 0.0);
```

Comment:
No need to comment on previous implementation

Reply:
Done